### PR TITLE
🔒️ Set JsonSerializerSettings' MaxDepth property to 128 & remove unused ComparisonMessage's utily methods and classes

### PR DESCRIPTION
**Pull request by PaulMathon**
NewtonSoft.Json package version < 13.0.1 exposes our apps to DoS attacks, this could be avoided by either upgrading the package or limiting the json max depth to 128

### CODE REVIEW

File: src/Criteo.OpenApi.Comparator/ComparisonMessage.cs [hunk 1]
```diff
@@ -126,48 +125,10 @@ private static string Location(IJsonDocument jsonDocument, IJsonLineInfo jsonTok
...
-        }
-
         /// <summary>
         /// Converts the currents ComparisonMessage object into a string message
         /// </summary>
         public override string ToString() =>
             $"code = {Code}, type = {Severity}, message = {Message}, docurl = {DocUrl}, mode = {Mode}";
     }
-
-    public class CustomComparer : IEqualityComparer<ComparisonMessage>
```
Comment (tibonihoo) [inline src/Criteo.OpenApi.Comparator/ComparisonMessage.cs:161]: I don't understand the link between the MaxDepth property change explained in the commit message and changing the way the ComparisonMessage would compare.

Could you explain in the commit message and if it's an unrelated clean-up move it to another commit ?

**Review by tibonihoo — CHANGES_REQUESTED** [review]

File: src/Criteo.OpenApi.Comparator/ComparisonMessage.cs (hunk 1 above)
Comment (PaulMathon) [reply src/Criteo.OpenApi.Comparator/ComparisonMessage.cs:161]: I guess I missed this file while splitting my commits, I'll update the message

**Review by tibonihoo — APPROVED** [review]
